security: TestRotateCerts failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/cbe3d10fd28f7a1a1025028b9f840e2f6e437133

Parameters:
```
TAGS=
GOFLAGS=
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=808732&tab=buildLog

```
=== RUN   TestRotateCerts
I180801 05:11:49.950094 13 security/certs.go:161  Generated CA key /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/ca.key
I180801 05:11:49.950704 13 security/certs.go:223  Wrote 1 certificates to /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/ca.crt
I180801 05:11:49.970293 13 security/certs.go:272  Generated node certificate: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/node.crt
I180801 05:11:49.970857 13 security/certs.go:278  Generated node key: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/node.key
I180801 05:11:49.983279 13 security/certs.go:392  Generated client certificate: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/client.root.crt
I180801 05:11:49.983367 13 security/certs.go:398  Generated client key: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/client.root.key
W180801 05:11:49.995328 13 server/status/runtime.go:287  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180801 05:11:50.011363 13 server/server.go:807  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180801 05:11:50.017119 13 server/config.go:554  [n?] 1 storage engine initialized
I180801 05:11:50.017149 13 server/config.go:557  [n?] RocksDB cache size: 128 MiB
I180801 05:11:50.017157 13 server/config.go:557  [n?] store 0: in-memory, size 0 B
I180801 05:11:50.021110 13 server/node.go:372  [n?] **** cluster dd855d54-c3ea-4ed0-842a-63558729e8b8 has been created
I180801 05:11:50.021180 13 server/server.go:1382  [n?] **** add additional nodes by specifying --join=127.0.0.1:38373
I180801 05:11:50.021445 13 gossip/gossip.go:363  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:38373" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:11 > 
I180801 05:11:50.022237 13 storage/store.go:1496  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180801 05:11:50.025598 13 server/node.go:461  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=6.9 KiB), ranges=1, leases=1, queries=0.00, writes=0.00, bytesPerReplica={p10=7043.00 p25=7043.00 p50=7043.00 p75=7043.00 p90=7043.00 pMax=7043.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I180801 05:11:50.025775 13 storage/stores.go:222  [n1] read 0 node addresses from persistent storage
I180801 05:11:50.025952 13 server/node.go:684  [n1] connecting to gossip network to verify cluster ID...
I180801 05:11:50.026056 13 server/node.go:709  [n1] node connected via gossip and verified as part of cluster "dd855d54-c3ea-4ed0-842a-63558729e8b8"
I180801 05:11:50.026118 13 server/node.go:533  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180801 05:11:50.026950 13 server/config.go:338  [n1] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180801 05:11:50.027057 13 server/server.go:1814  [n1] Could not start heap profiler worker due to: directory to store profiles could not be determined
I180801 05:11:50.028688 13 server/server.go:1519  [n1] starting https server at 127.0.0.1:42823
I180801 05:11:50.028727 13 server/server.go:1520  [n1] starting grpc/postgres server at 127.0.0.1:38373
I180801 05:11:50.028745 13 server/server.go:1521  [n1] advertising CockroachDB node at 127.0.0.1:38373
W180801 05:11:50.028771 13 sql/jobs/registry.go:299  [n1] unable to get node liveness: node not in the liveness table
I180801 05:11:50.047850 309 sql/event_log.go:126  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I180801 05:11:50.065931 357 sql/event_log.go:126  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:root}
I180801 05:11:50.073686 363 sql/event_log.go:126  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I180801 05:11:50.077087 278 storage/replica_command.go:279  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180801 05:11:50.082499 299 storage/replica_command.go:279  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I180801 05:11:50.088227 29 storage/replica_command.go:279  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180801 05:11:50.095403 393 storage/replica_command.go:279  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
I180801 05:11:50.107723 407 sql/event_log.go:126  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:gen_random_uuid()::STRING User:root}
I180801 05:11:50.109449 374 storage/replica_command.go:279  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I180801 05:11:50.131782 413 storage/replica_command.go:279  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180801 05:11:50.145880 381 storage/replica_command.go:279  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
W180801 05:11:50.148796 378 storage/intent_resolver.go:642  [n1,s1] failed to push during intent resolution: failed to push "sql txn" id=75822f2d key=/Table/SystemConfigSpan/Start rw=true pri=0.00836472 iso=SERIALIZABLE stat=PENDING epo=0 ts=1533100310.139800412,0 orig=1533100310.139800412,0 max=1533100310.139800412,0 wto=false rop=false seq=2
I180801 05:11:50.151265 434 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I180801 05:11:50.152883 425 storage/replica_command.go:279  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180801 05:11:50.157805 439 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I180801 05:11:50.159139 473 storage/replica_command.go:279  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180801 05:11:50.159530 13 server/server.go:1603  [n1] done ensuring all necessary migrations have run
I180801 05:11:50.159566 13 server/server.go:1606  [n1] serving sql connections
I180801 05:11:50.165930 488 server/server_update.go:67  [n1] no need to upgrade, cluster already at the newest version
I180801 05:11:50.167051 490 sql/event_log.go:126  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:38373} Attrs: Locality: ServerVersion:2.0-11} ClusterID:dd855d54-c3ea-4ed0-842a-63558729e8b8 StartedAt:1533100310026083260 LastUp:1533100310026083260}
I180801 05:11:50.169819 340 storage/replica_command.go:279  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180801 05:11:50.174277 349 storage/replica_command.go:279  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180801 05:11:50.179914 525 storage/replica_command.go:279  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180801 05:11:50.186414 534 storage/replica_command.go:279  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180801 05:11:50.193956 448 storage/replica_command.go:279  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180801 05:11:50.198805 562 storage/replica_command.go:279  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
W180801 05:11:50.205354 578 storage/intent_resolver.go:642  [n1,s1] failed to push during intent resolution: failed to push "split" id=83716227 key=/Local/Range/Table/18/RangeDescriptor rw=true pri=0.06184289 iso=SERIALIZABLE stat=PENDING epo=0 ts=1533100310.200322791,1 orig=1533100310.198822432,0 max=1533100310.198822432,0 wto=false rop=false seq=1
I180801 05:11:50.214057 430 storage/replica_command.go:279  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
W180801 05:11:50.237052 587 storage/intent_resolver.go:642  [n1,s1] failed to push during intent resolution: failed to push "split" id=70a545cd key=/Local/Range/Table/19/RangeDescriptor rw=true pri=0.01778221 iso=SERIALIZABLE stat=PENDING epo=0 ts=1533100310.214074592,0 orig=1533100310.214074592,0 max=1533100310.214074592,0 wto=false rop=false seq=1
W180801 05:11:50.238191 589 storage/intent_resolver.go:642  [n1,s1] failed to push during intent resolution: failed to push "split" id=70a545cd key=/Local/Range/Table/19/RangeDescriptor rw=true pri=0.01778221 iso=SERIALIZABLE stat=PENDING epo=0 ts=1533100310.214074592,0 orig=1533100310.214074592,0 max=1533100310.214074592,0 wto=false rop=false seq=1
I180801 05:11:50.239668 594 storage/replica_command.go:279  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180801 05:11:50.244398 556 storage/replica_command.go:279  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180801 05:11:50.250881 629 storage/replica_command.go:279  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180801 05:11:50.257793 643 storage/replica_command.go:279  [split,n1,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/50 [r21]
I180801 05:11:50.263826 462 storage/replica_command.go:279  [split,n1,s1,r21/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I180801 05:11:50.313607 13 security/certs.go:161  Generated CA key /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/ca.key
I180801 05:11:50.314121 13 security/certs.go:223  Wrote 1 certificates to /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/ca.crt
I180801 05:11:50.324885 13 security/certs.go:272  Generated node certificate: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/node.crt
I180801 05:11:50.324958 13 security/certs.go:278  Generated node key: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/node.key
I180801 05:11:50.333968 13 security/certs.go:392  Generated client certificate: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/client.root.crt
I180801 05:11:50.334113 13 security/certs.go:398  Generated client key: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/client.root.key
E180801 05:11:50.338082 694 server/server.go:1627  [n1,client=127.0.0.1:48202] remote error: tls: bad certificate
E180801 05:11:50.356726 676 server.go:2921  http: TLS handshake error from 127.0.0.1:56784: remote error: tls: bad certificate
I180801 05:11:50.366191 20 security/certificate_manager.go:183  received signal "hangup", triggering certificate reload
I180801 05:11:50.366264 42 util/log/clog.go:598  hangup received, flushing logs
W180801 05:11:50.366633 699 server/server.go:1863  [n1] error closing gzip response writer: write tcp 127.0.0.1:42823->127.0.0.1:56788: write: broken pipe
I180801 05:11:50.372063 20 security/certificate_manager.go:187  successfully reloaded certificates
E180801 05:11:50.380805 670 server.go:2921  http: TLS handshake error from 127.0.0.1:56800: remote error: tls: bad certificate
W180801 05:11:50.385077 723 server/server.go:1863  [n1] error closing gzip response writer: write tcp 127.0.0.1:42823->127.0.0.1:56802: write: connection reset by peer
I180801 05:11:50.399421 13 security/certs.go:161  Generated CA key /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/ca.key
I180801 05:11:50.401193 13 security/certs.go:209  Found 1 certificates in /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/ca.crt
I180801 05:11:50.401307 13 security/certs.go:223  Wrote 2 certificates to /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/ca.crt
I180801 05:11:50.440285 13 security/certs.go:272  Generated node certificate: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/node.crt
I180801 05:11:50.440671 13 security/certs.go:278  Generated node key: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/node.key
I180801 05:11:50.464361 13 security/certs.go:392  Generated client certificate: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/client.root.crt
I180801 05:11:50.471656 13 security/certs.go:398  Generated client key: /go/src/github.com/cockroachdb/cockroach/artifacts/certs_test128761880/client.root.key
E180801 05:11:50.481774 737 server/server.go:1627  [n1,client=127.0.0.1:48226] tls: failed to verify client's certificate: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "Cockroach CA")
--- FAIL: TestRotateCerts (0.55s)
	certs_rotation_test.go:161: issuing SIGHUP
	certs_rotation_test.go:226: issuing SIGHUP
	certs_rotation_test.go:242: remote error: tls: bad certificate

```

--- a/pkg/security/certs_rotation_test.go
+++ b/pkg/security/certs_rotation_test.go
@@ -228,17 +228,15 @@ func TestRotateCerts(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// Wait until client3 succeeds.
+	// Wait until client3 succeeds (both http and sql).
 	testutils.SucceedsSoon(t,
 		func() error {
 			if err := clientTest(thirdClient); err != nil {
-				return errors.Errorf("third client failed: %v", err)
+				return errors.Errorf("third HTTP client failed: %v", err)
+			}
+			if _, err := thirdSQLClient.Exec("SELECT 1"); err != nil {
+				return errors.Errorf("third SQL client failed: %v", err)
 			}
 			return nil
 		})
-
-	// SQL client 3 should also succeed.
-	if _, err := thirdSQLClient.Exec("SELECT 1"); err != nil {
-		t.Fatal(err)
-	}
 }

